Using The RESTEasy Tracing Feature In WildFly

liweinan opened the pull request.

liweinan commented:
@jamezp @rsearls This is still a draft, and I'll finish writing this blog post this week. Because there are already blog posts and document describing the tracing feature itself, so the focus is on the tracing feature usage in WildFly. If you have any suggestions on this article please comment and I'll do the modifications :D

jamezp commented on _posts/2023-04-12-tracing-feature-in-wildfly.md:
```diff
@@ -0,0 +1,198 @@
+
+The RESTEasy tracing feature has been integrated into WildFly since its version 28. The usage of the tracing feature in RESTEasy is already introduced in RESTEasy Documentation[^resteasydoc] and some blog posts[^resteasyposts]. In this article, I’ll focus on the usage of this feature in WildFly.
+
+There is an example in the `resteasy-example`[^resteaasyexample] project showing the usage of the tracing feature, and it uses the Galleon Feature Pack to provision a WildFly 28 server, so I’ll use this example this this article.
```
I would say "it uses Galleon" here.

liweinan replied:
okay!

jamezp reviewed (commented):
This is just a general comment about the example itself and something I think I'll look at when I make a few small changes to how the example provisions a server.

I wonder if we should create a `header` resource which would show how the headers could be used too. This would show either a JSON response or plain text for the header so the user doesn't have to leave the browser to see the logs. I do think keeping this blog post focused on the logs is fine though.

rsearls commented:
With those minor changes it looks fine.

liweinan commented on _posts/2023-04-12-tracing-feature-in-wildfly.md:
````diff
@@ -0,0 +1,198 @@
+```
+
+```bash
+$ mvn wildfly:run
````
yes sir!

liweinan commented on _posts/2023-04-12-tracing-feature-in-wildfly.md:
````diff
@@ -0,0 +1,198 @@
+```
+
+```bash
+➤ ./jboss-cli.sh
````
cool!

liweinan commented on _posts/2023-04-12-tracing-feature-in-wildfly.md:
````diff
@@ -0,0 +1,198 @@
+Now we can change the `tracing-type` from `OFF` to `ON` on the fly:
+
+```bash
+[standalone@localhost:9990 /] cd subsystem=jaxrs
````
I'll modify it :D

liweinan commented on _posts/2023-04-12-tracing-feature-in-wildfly.md:
````diff
@@ -0,0 +1,198 @@
+
+```bash
+[standalone@localhost:9990 subsystem=jaxrs] cd /
+[standalone@localhost:9990 /] :reload
````
okay!

liweinan commented:
> This is just a general comment about the example itself and something I think I'll look at when I make a few small changes to how the example provisions a server.
> 
> I wonder if we should create a `header` resource which would show how the headers could be used too.

Could you please help to provide this in the `resteasy-example` side so I could catch your thought? And I'll update this blog post accordingly :D

> This would show either a JSON response or plain text for the header so the user doesn't have to leave the browser to see the logs. I do think keeping this blog post focused on the logs is fine though.

liweinan commented:
@jamezp @rsearls Thanks for reviewing this post! I'll update the text accordingly.

liweinan commented:
> > This is just a general comment about the example itself and something I think I'll look at when I make a few small changes to how the example provisions a server.

> > I wonder if we should create a `header` resource which would show how the headers could be used too.
> 
> @jamezp Could you please help to provide this in the `resteasy-example` side so I could catch your thought? And I'll update this blog post accordingly :D

Guess I understand your meaning now. I'll provide a header resource in the example :D
> 
> > This would show either a JSON response or plain text for the header so the user doesn't have to leave the browser to see the logs. I do think keeping this blog post focused on the logs is fine though.

liweinan commented:
Relative PR: https://github.com/resteasy/resteasy-examples/pull/123

liweinan commented:
@jamezp I have updated the blog post, and create a PR to the `tracing-example` to add notes about JSON format header. I moved this PR to `ready for review`, could you please help to check if anything else need to be modified? Thanks!

jamezp commented on _posts/2023-04-12-tracing-feature-in-wildfly.md:
```diff
@@ -0,0 +1,205 @@
+
+[^galleon]: [Galleon Provisioning](https://github.com/wildfly/galleon)
+
+[^wildflytracing]: [tracingEnabled in WildFly](https://github.com/wildfly/wildfly/pull/15198/files#diff-6716f11308b916188cda71be70b388833fb56713df3644b1f4a41fa4815c8275R230)
```
I wonder if this should link the documentation itself https://github.com/wildfly/wildfly/pull/15198/files#diff-b1b3a12a4b68c3397dacaab84d50790141d87677f46552c5138f520312aa3782R155.

jamezp commented on _posts/2023-04-12-tracing-feature-in-wildfly.md:
```diff
@@ -0,0 +1,205 @@
+layout:     post
+title:      "Using The RESTEasy Tracing Feature In WildFly"
+subtitle:   ""
+date:       2023-04-12
```
We should change this and the file name IMO before we merge this.

jamezp commented on _posts/2023-04-12-tracing-feature-in-wildfly.md:
````diff
@@ -0,0 +1,205 @@
+The above command will build the example, download necessary WildFly components. Then run the provisioned WildFly server with the following command:
+
+```bash
+$ mvn wildfly:dev
````
You might want to use `mvn wildfly:provision wildfly:dev` here. If a previous `install` was not done, the `jboss-home` won't be found.

jamezp commented on _posts/2023-04-12-tracing-feature-in-wildfly.md:
````diff
@@ -0,0 +1,205 @@
+"process-state" => "reload-required"
+```
+
+So we need to use the following command to reload the WildFly configuration:
````
"So we need to use the following command to reload the WildFly services:". "configuration" doesn't sound right here to me.

jamezp commented on _posts/2023-04-12-tracing-feature-in-wildfly.md:
```diff
@@ -0,0 +1,205 @@
+
+[^resteasyposts]: [A brief introduction to the RESTEasy Tracing Feature](https://resteasy.dev/2018/06/11/a-brief-introduction-to-the-resteasy-tracing-feature/) / [RESTEasy Tracing Feature Now - Supports JSON formatted information](https://resteasy.dev/2018/09/05/resteasy-tracing-feature-now-supports-json-formatted-information/)
+
+[^resteaasyexample]: https://github.com/resteasy/resteasy-examples/pull/122
```
Shouldn't this just point to https://github.com/resteasy/resteasy-examples/tree/main/tracing-example

liweinan commented:
@jamezp thanks for the review! I'll finish writing this blog post by this week.

liweinan commented:
@jamezp I have finished updating the text, could you please help to do a review? Thanks!
